x-pack/plugins/enterprise_search/server/utils/create_ml_inference_pipeline.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+
+import { formatMlPipelineBody } from './create_pipeline_definitions';
+
+export interface CreatedPipeline {
+  created: string;
+}
+
+// TODO: document
+export const createMlInferencePipeline = async (
+  pipelineName: string,
+  modelId: string,
+  sourceField: string,
+  destinationField: string,
+  esClient: ElasticsearchClient
+): Promise<CreatedPipeline> => {
+  const inferencePipelineGeneratedName = `ml-inference-${pipelineName}`;
```

Comment:
I added this to the frontend code, but I can remove it, as I like it here better.

Reply:
Yeah, I think it's better to generate this on the backend so the prefix guaranteed.